apps/video/js/video.js

```diff
@@ -108,6 +108,27 @@ var loadingChecker =
   new VideoLoadingChecker(dom.player, dom.inUseOverlay, dom.inUseOverlayTitle,
                           dom.inUseOverlayText);
 
+var lastPlayTime = 0;
+
+// Workaround to fix issue reported in Bug 1214157
+// We would like to pause video playback when interrupted by competing audio.
+// Audio Channel API throws interruptbegin event after play event (Bug 1192748)
+// If performance.now() - lastPlayTime is small number we ignore interruptbegin
+// and pause video playback only when interrupted by competing audio.
+// This workaround should be removed with the fix of Bug 1206581
+dom.player.addEventListener('play', function() {
+  lastPlayTime = window.performance.now();
+});
+
+dom.player.addEventListener('mozinterruptbegin', function() {
+  // Ignore mozinterruptbegin event if it immediately follows play event
+  if ((window.performance.now() - lastPlayTime) > 50) {
+    // Pause video when interrupted by competing audio.
+    pause();
+  }
+  lastPlayTime = 0;
```

Comment:
I was using lastPlayTime flag more as indication of getting inetrruptbegin event immediately after play event and handling it versus the actual last play event received time.
Having said that when 'mozinterruptbegin' is fired in response to competing audio, 'now - lastPlayTime' difference is expected to be greater than 50ms even if 'lastPlayTime' didn't get reset, so I have updated patch to reset lastPlayTime only when interrupted i.e.  interruptbegin from audio competing is received.